source/binding/OperatorExpressions.cpp

```diff
@@ -1620,19 +1748,29 @@ void StreamingConcatenationExpression::serializeTo(ASTSerializer& serializer) co
     serializer.write("sliceSize", sliceSize);
     if (!streams().empty()) {
         serializer.startArray("streams");
-        for (auto op : streams())
-            serializer.serialize(*op);
+        for (auto op : streams()) {
+            serializer.startObject();
+            serializer.write("operand", op->operand);
+            if (op->with) {
+                serializer.write("withKind", toString(op->with->kind));
+                serializer.write("left", op->with->left);
```

Comment:
You're calling these with pointer values... I wonder if this is picking the write<T> overloads that are intended for integers? That seems not good.

Reply:
I fix the bug by dereferencing the pointers.
```c++
            serializer.write("operand", *op->operand);
            if (op->with) {
                serializer.write("withKind", toString(op->with->kind));
                serializer.write("left", *op->with->left);
                if (op->with->right)
                    serializer.write("right", *op->with->right);
            }
```